src/Mod/Part/App/ExtrusionHelper.h

```diff
@@ -0,0 +1,54 @@
+/***************************************************************************
+ *   Copyright (c) 2022 Uwe St�hr <uwestoehr@lyx.org>                      *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef PART_EXTRUSIONHELPER_H
+#define PART_EXTRUSIONHELPER_H
+
+#include <list>
+#include <gp_Dir.hxx>
+#include <TopoDS_Shape.hxx>
+
+namespace Part
+{
+
+class PartExport ExtrusionHelper
+{
+public:
+    ExtrusionHelper();
+
+    /**
+     * @brief makeDraft: creates a drafted extrusion shape out of the input 2D shape
+     */
+    static void makeDraft(const TopoDS_Shape& shape,

```

Comment:
IMO, the class ExtrusionParameters in the original code was a good thing because it kept the list of arguments very small. Thus, I recommend to move the class ExtrusionParameters from Extrusion to this class but I would drop the "Extrusion" and just call it Parameters.

Reply:
Yes and I wanted to keep this. I failed because in Line 648 ff. I need to call the Part::ExtrusionHelper::makeDraft in 4 different ways. 
Therefore the many arguments are looking weird but I don't know a better way. Do you have an idea?